### PR TITLE
Revamp documentation

### PR DESCRIPTION
**Pull request by saulecabrera**
The objective of this commit is to:

* Bring the documentation up to date, by removing/correcting the stale pieces
* Introduce a new organzation approach to documentation making it easier to evolve it over time.
* Reduce the text-density of the README

In general this commit strives to add as little new information as possible and primarily re-organizes the existing documentation. The only extra documentation is the JavaScript API support, which was lacking as correctly noted in https://github.com/bytecodealliance/javy/issues/749#issuecomment-2340448398

## Checklist

- [x] I've updated the relevant CHANGELOG files if necessary. Changes to `javy-cli` and `javy-core` do not require updating CHANGELOG files.
- [x] I've updated the relevant crate versions if necessary. [Versioning policy for library crates](https://github.com/bytecodealliance/javy/blob/main/docs/contributing.md#versioning-for-library-crates)
- [x] I've updated documentation including crate documentation if necessary.

### CODE REVIEW

File: docs/docs-contributing-developing.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (jeffcharles) [inline docs/docs-contributing-developing.md:1]: Nit:

```suggestion
```

File: docs/docs-contributing-testing-locally.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+```
+cargo +stable install cargo-hack --locked
+```
+
+3. Run tests, eg:
+
+```
+cargo +stable install cargo-hack --locked
+```
+
```
Comment (jeffcharles) [inline docs/docs-contributing-testing-locally.md:21]:
```suggestion
```

File: docs/docs-using-js-api-support.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# JavaScript API Support
+
+Javy by default supports all the JavaScript features and APIs provided by
+QuickJS, plus partial support for additional APIs.
```
Comment (jeffcharles) [inline docs/docs-using-js-api-support.md:4]: Nit: Might be worth also saying this includes ES2023 APIs but not NodeJS APIs.

File: README.md [hunk 4]
```diff
@@ -10,69 +10,23 @@
   <p>
     <a href="https://github.com/bytecodealliance/javy/actions/workflows/ci.yml"><img alt="Build status" src="https://github.com/bytecodealliance/javy/actions/workflows/ci.yml/badge.svg?branch=main" /></a>
     <a href="https://bytecodealliance.zulipchat.com/#narrow/stream/370816-javy"><img src="https://img.shields.io/badge/zulip-join_chat-brightgreen.svg" alt="zulip chat" /></a>
-  </p>
+  </p>
```
Comment (jeffcharles) [inline README.md:13]: Nit:

```suggestion
  </p>
```

**Review by jeffcharles — APPROVED** [review]
Looks good!

Any reason for the leading `docs-` on the filenames in the `docs` directory?

Only change I'd really like to see made is the one to remove the duplicate `cargo +stable install cargo-hack --locked`, others are all nits.

**Comment by saulecabrera** [comment]
> Any reason for the leading `docs-` on the filenames in the `docs` directory?

No strong reason. I wanted to make sure that we can easily move these files around if need be  (independent of the directory) allowing us to fully disambiguate them from other markdown files (if case we introduce new ones).

**Comment by saulecabrera** [comment]
I'm not sure why the jobs seem to be stuck. I'll close and re-open.

**Pull request merged**